### DISCUSSION TITLE
Documentation addon for HTMX.  HTMX hx-on and surreal equivalent.

### DISCUSSION THREAD

**Discussion by figuerom16** [gnat/surreal#18] [Ideas]
Something that isn't represented in the README is that surreal can function as a replacement for any `hx-on::<event>`
https://htmx.org/attributes/hx-on/

For example:

HTMX:
```javascript
<button hx-get="/" hx-on::afterRequest="console.log('run script')">TEST</button>
```
Surreal:
```javascript
<button hx-get="/">TEST
	<script>
		me().on('htmx:afterRequest', _=>{
			console.log('run script')
		})
	</script>
</button>
```

Maybe it just wasn't obvious to me, but being able to use Surreal's LoB with HTMX events is a massive win to me.
https://htmx.org/events/

**Comment by gnat** [comment]
It's a cool idea, and I agree. Will likely add an htmx patterns section in the near future. I really appreciate the feedback.
